src/test/miner_tests.cpp

```diff
@@ -291,125 +384,158 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
+    BOOST_CHECK(pblocktemplate = BlockAssembler(chainparams).CreateNewBlock(scriptPubKey));
+    // only 211 transactions including coinbase, not 212
+    BOOST_CHECK_EQUAL(211, pblocktemplate->block.vtx.size());
+    BOOST_CHECK_GE(2000000, ::GetSerializeSize(pblocktemplate->block, SER_NETWORK, PROTOCOL_VERSION));
+    delete pblocktemplate;
     mempool.clear();
 
-    // child with higher feerate than parent
     tx.vin[0].scriptSig = CScript() << OP_1;
-    tx.vin[0].prevout.hash = txFirst[1]->GetHash();
-    tx.vout[0].nValue = BLOCKSUBSIDY-HIGHFEE;
-    hash = tx.GetHash();
-    mempool.addUnchecked(hash, entry.Fee(HIGHFEE).Time(GetTime()).SpendsCoinbase(true).FromTx(tx));
-    tx.vin[0].prevout.hash = hash;
-    tx.vin.resize(2);
-    tx.vin[1].scriptSig = CScript() << OP_1;
-    tx.vin[1].prevout.hash = txFirst[0]->GetHash();
-    tx.vin[1].prevout.n = 0;
-    tx.vout[0].nValue = tx.vout[0].nValue+BLOCKSUBSIDY-HIGHERFEE; //First txn output + fresh coinbase - new txn fee
-    hash = tx.GetHash();
-    mempool.addUnchecked(hash, entry.Fee(HIGHERFEE).Time(GetTime()).SpendsCoinbase(true).FromTx(tx));
-    BOOST_CHECK(pblocktemplate = BlockAssembler(chainparams).CreateNewBlock(scriptPubKey));
-    delete pblocktemplate;
+
+    // Test with an orphan tx in the mempool. Note that this is testing a case that should never occur,
+    // because AcceptToMemoryPool would reject the orphan. Using addUnchecked bypasses that rejection.
+    PrepareTemplate(2, MINIMUM_FEE, [&](size_t i, auto& entry, auto& tx) {
+        if (i != 0) mempool.addUnchecked(tx.GetHash(), entry.SigOps(1).FromTx(tx));
+    });
+    BOOST_CHECK_EXCEPTION(BlockAssembler(chainparams).CreateNewBlock(scriptPubKey), std::runtime_error, err_is("bad-txns-inputs-missingorspent"));
     mempool.clear();
 
+    // child with higher priority than parent
+    // TODO: re-enable this when we have implemented priority logic for ZIP 317 block construction.
+
+    // const CAmount HIGHFEE = MINIMUM_FEE*2;
+    // const CAmount HIGHERFEE = MINIMUM_FEE*3;
+    //
+    // tx.vin[0].scriptSig = CScript() << OP_1;
+    // tx.vin[0].prevout.hash = CoinbaseTx(1)->GetHash();
+    // tx.vin[0].prevout.n = 0;
+    // tx.vout[0].nValue = MinerSubsidy(1) - HIGHFEE;
+    // hash = tx.GetHash();
+    // mempool.addUnchecked(hash, entry.Fee(HIGHFEE).Time(GetTime()).SpendsCoinbase(true).FromTx(tx));
+    // tx.vin[0].prevout.hash = hash;
+    // tx.vin.resize(2);
+    // tx.vin[1].scriptSig = CScript() << OP_1;
+    // tx.vin[1].prevout.hash = CoinbaseTx(2)->GetHash();
+    // tx.vin[1].prevout.n = 0;
+    // tx.vout[0].nValue = tx.vout[0].nValue + MinerSubsidy(2) - HIGHERFEE; // First txn output + fresh coinbase - new txn fee
+    // hash = tx.GetHash();
+    // mempool.addUnchecked(hash, entry.Fee(HIGHERFEE).Time(GetTime()).SpendsCoinbase(true).FromTx(tx));
+    // BOOST_CHECK(pblocktemplate = BlockAssembler(chainparams).CreateNewBlock(scriptPubKey));
+    // delete pblocktemplate;
+    // mempool.clear();
```

Comment:
I guess we should probably re-enable this.

Reply:
Oh, actually it probably no longer makes sense. There's no reason why a child paying a higher fee than the parent would make any difference that we can test deterministically. So we should delete this.